src/dawn/IIR/MetaInformation.h

```diff
@@ -0,0 +1,493 @@
+
+  /// This is a set of AccessIDs which correspond to fields. This allows to fully identify if a
+  /// AccessID is a field, variable or literal as literals have always strictly negative IDs and
+  /// variables are neither field nor literals.
+  std::set<int> FieldAccessIDSet_;
+
+  /// This is an ordered list of IDs of fields that belong to the user API call of the program
+  std::vector<int> apiFieldIDs_;
+
+  /// Set containing the AccessIDs of fields which are represented by a temporary storages
+  std::set<int> TemporaryFieldAccessIDSet_;
+
+  /// Set containing the AccessIDs of fields which are manually allocated by the stencil and serve
+  /// as temporaries spanning over multiple stencils
+  std::set<int> AllocatedFieldAccessIDSet_;
+
+  /// Set containing the AccessIDs of "global variable" accesses. Global variable accesses are
+  /// represented by global_accessor or if we know the value at compile time we do a constant
+  /// folding of the variable
+  std::set<int> GlobalVariableAccessIDSet_;
+
+  /// Map of AccessIDs to the list of all AccessIDs of the multi-versioned field, for fields and
+  /// variables
+  VariableVersions variableVersions_;
+
+  /// Stencil description statements. These are built from the StencilDescAst of the sir::Stencil
+  std::vector<std::shared_ptr<Statement>> stencilDescStatements_;
+  std::unordered_map<std::shared_ptr<StencilCallDeclStmt>, int> StencilCallToStencilIDMap_;
+  std::unordered_map<int, std::shared_ptr<StencilCallDeclStmt>> IDToStencilCallMap_;
+
+  /// StageID to name Map. Filled by the `PassSetStageName`.
+  std::unordered_map<int, std::string> StageIDToNameMap_;
+
+  /// Referenced stencil functions in this stencil (note that nested stencil functions are not
+  /// stored here but rather in the respecticve `StencilFunctionInstantiation`)
+  std::vector<std::shared_ptr<StencilFunctionInstantiation>> stencilFunctionInstantiations_;
+  std::unordered_map<std::shared_ptr<StencilFunCallExpr>,
+                     std::shared_ptr<StencilFunctionInstantiation>>
+      ExprToStencilFunctionInstantiationMap_;
+
+  /// lookup table containing all the stencil function candidates, whose arguments are not yet bound
+  std::unordered_map<std::shared_ptr<StencilFunctionInstantiation>,
+                     StencilFunctionInstantiationCandidate>
+      stencilFunInstantiationCandidate_;
+
+  /// BoundaryConditionCall to Extent Map. Filled my `PassSetBoundaryCondition`
+  std::unordered_map<std::shared_ptr<BoundaryConditionDeclStmt>, Extents>
+      BoundaryConditionToExtentsMap_;
+
+  /// Field Name to BoundaryConditionDeclStmt
+  std::unordered_map<std::string, std::shared_ptr<BoundaryConditionDeclStmt>>
+      FieldnameToBoundaryConditionMap_;
+
+  /// Set of all the IDs that are locally cached
+  std::set<int> CachedVariableSet_;
+
+  /// Map of Field ID's to their respecive legal dimensions for offsets if specified in the code
+  std::unordered_map<int, Array3i> fieldIDToInitializedDimensionsMap_;
```

Comment:
the naming here I think is confusing

Reply:
why, these are the dimensions that the user initialized? what would be your suggetstion?